File: CPP/No421.cpp
/**
 * Created by Xiaozhong on 2020/9/18.
 * Copyright (c) 2020/9/18 Xiaozhong. All rights reserved.
 */
#include "iostream"
#include "vector"
#include "string"
#include "unordered_set"

using namespace std;

class Solution {
public:
    int findMaximumXOR(vector<int> &nums) {
        int ans = 0;
        int mask = 0;
        for (int i = 30; i >= 0; --i) {
            mask = mask | (1 << i);
            unordered_set<int> set;
            for (int num : nums) {
                set.insert(num & mask);
            }
            int temp = ans | (1 << i);
            for (int prefix : set) {
                if (set.count(prefix ^ temp)) {
                    ans = temp;
                    break;
                }
            }
        }
        return ans;
    }
};

File: Java/src/No421.java
/*
 * Copyright (c) 2021, HuXiaozhong. All rights reserved.
 * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
 */

public class No421 {
    public int findMaximumXOR(int[] nums) {
        int ans = 0;
        for (int left = 0; left < nums.length; ++left) {
            for (int right = left + 1; right < nums.length; ++right) {
                ans = Math.max(nums[left] ^ nums[right], ans);
            }
        }
        return ans;
    }
}
